api/client/src/main/java/org/projectnessie/client/rest/ResponseCheckFilter.java

```diff
@@ -113,14 +114,23 @@ private static NessieError decodeErrorObject(
         JsonNode errorData = reader.readTree(capturing);
         try {
           error = reader.treeToValue(errorData, NessieError.class);
+        } catch (MismatchedInputException e) {
+          // If the error payload is valid JSON, but does not represent a NessieError, it is likely
+          // produced by Quarkus and contains the server-side logged error ID. Report the raw JSON
+          // text to the caller for trouble-shooting.
+          error =
+              ImmutableNessieError.builder()
+                  .message(capturing.captured())
+                  .status(status.getCode())
+                  .reason(status.getReason())
+                  .build();
         } catch (JsonProcessingException e) {
-
           // If the error payload is valid JSON, but does not represent a NessieError, it is likely
           // produced by Quarkus and contains the server-side logged error ID. Report the raw JSON
           // text to the caller for trouble-shooting.
           error =
               ImmutableNessieError.builder()
-                  .message(errorData.toString())
+                  .message(capturing.captured())
                   .status(status.getCode())
                   .reason(status.getReason())
                   .clientProcessingException(e)
```

Comment:
Would it make sense to keep client processing exceptions in the `NessieError` object, but use a simpler message instead of `getFullMessage()` in exceptions? This way, the parsing error is still available in case it is required for investigation.

If that is too much, how about not populating `clientProcessingException()` at all?

Reply:
That is to say, I do not see much difference in these two exception cases (and line 147 for that matter).

Reply:
There is some nuanced difference. Added a comment.

Reply:
Previously the idea was that we'd include the JSON text in the message in this case, but since we introduced input capturing, I think have separate catch clauses for JSON exception does not add value anymore.